chainstate/tx-verifier/src/transaction_verifier/input_check.rs

```diff
@@ -0,0 +1,301 @@
+// Copyright (c) 2024 RBB S.r.l
+// opensource@mintlayer.org
+// SPDX-License-Identifier: MIT
+// Licensed under the MIT License;
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// https://github.com/mintlayer/mintlayer-core/blob/master/LICENSE
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use chainstate_types::block_index_ancestor_getter;
+use common::{
+    chain::{
+        block::timestamp::BlockTimestamp,
+        signature::{verify_signature, Signable, Transactable},
+        ChainConfig, GenBlock, TxInput, TxOutput, UtxoOutPoint,
+    },
+    primitives::{BlockHeight, Id},
+};
+use itertools::Itertools;
+
+use crate::TransactionVerifierStorageRef;
+
+use super::{
+    error::ConnectTransactionError, signature_destination_getter::SignatureDestinationGetter,
+    TransactionSourceForConnect,
+};
+
+pub struct BlockVerificationContext<'a> {
+    chain_config: &'a ChainConfig,
+    destination_getter: SignatureDestinationGetter<'a>,
```

Comment:
Note that after introducing scripting, this will probably be eliminated. This is because SignatureDestinationGetter assumes that there's exactly one signature to verify per input, while in reality, it can be more complex than just that.